typing_extensions/src_py3/typing_extensions.py

```diff
@@ -0,0 +1,612 @@
+import abc
+import collections
+import contextlib
+import sys
+import typing
+try:
+    import collections.abc as collections_abc
+except ImportError:
+    import collections as collections_abc  # Fallback for PY3.2.
+
+if hasattr(typing, '_generic_new'):
+    _generic_new = typing._generic_new
+else:
+    def _generic_new(base_cls, cls, *args, **kwargs):
+        return base_cls.__new__(cls, *args, **kwargs)
```

Comment:
I am not 100% sure this is a safe substitution. Do you have a test for this? IIRC, the point was that ``__init__`` was not called at instantiation of subscripted generics.